### PR TITLE
Fix: remove shared security rule when no svc with shared-nsg annotation exists

### PR DESCRIPTION
**Pull request by MartinForReal**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:
remove shared security rule when no svc with shared-nsg annotation exists

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Fix: remove unsed shared nsg rule when there is no svc having azure-shared-securityrule annotations set
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by feiskyer** [comment]
Could you add an e2e test for the issue?

**Comment by MartinForReal** [comment]
https://github.com/kubernetes-sigs/cloud-provider-azure/blob/7a494eba9870c1719a3fcd5a1e1bd442d75ff4e4/tests/e2e/network/network_security_group.go#L175 there is already one test case for this scenario.

File: pkg/provider/azure_loadbalancer.go [hunk 1]
```diff
@@ -2664,10 +2665,14 @@ func (az *Cloud) reconcileSecurityRules(sg network.SecurityGroup, service *v1.Se
 		}
 		if foundRule && allowsConsolidation(expectedRule) {
 			index, _ := findConsolidationCandidate(updatedRules, expectedRule)
-			updatedRules[index] = consolidate(updatedRules[index], expectedRule)
+			if updatedRules[index].DestinationAddressPrefixes != nil {
+				updatedRules[index] = consolidate(updatedRules[index], expectedRule)
+			} else {
+				updatedRules = append(updatedRules[:index], updatedRules[index+1:]...)
```
Comment (feiskyer) [inline pkg/provider/azure_loadbalancer.go:2671]: Is the rule delete here required? The newly added L2624 should remove the unwanted rule.

Comment (MartinForReal) [reply pkg/provider/azure_loadbalancer.go:2671]: L2624 will be triggered only when there is another expected rule. if there is only one rule in expected rules, L2624 will not be executed.

Comment (feiskyer) [reply pkg/provider/azure_loadbalancer.go:2671]: OK, thanks for clarification!

**Review by feiskyer — APPROVED** [review]
LGTM

File: pkg/provider/azure_loadbalancer.go [hunk 2]
```diff
@@ -2621,6 +2621,7 @@ func (az *Cloud) reconcileSecurityRules(sg network.SecurityGroup, service *v1.Se
 				}
 				if sharedRule.DestinationAddressPrefixes == nil {
 					klog.V(4).Infof("Didn't find DestinationAddressPrefixes in shared rule for service %s", service.Name)
+					updatedRules = append(updatedRules[:sharedIndex], updatedRules[sharedIndex+1:]...)
```
Comment (nilo19) [inline pkg/provider/azure_loadbalancer.go:2624]: Why do we need to include it here and remove it later in L2671?

Comment (MartinForReal) [reply pkg/provider/azure_loadbalancer.go:2624]: we removed the rules here..

**Comment by MartinForReal** [comment]
e2e case is modified.

**Pull request merged**